developer_guides/testbench/debug_in_testbench.rst

```diff
@@ -0,0 +1,171 @@
+window and place a break point there with right mouse button (a red
+stop sign). If there are issues those happen at topology parsing or
+ẃithin component in instantiation in new() place the breakpoint to
+parse_topology() line.
+
+.. figure:: fig_ddd.png
+
+	    The ddd debugger start view.
+
+.. figure:: fig_add_breakpoint.png
+
+	    Breakpoint added with right mouse click.
+	    
+The breakpoint is placed after topology parsing since the component
+symbols do not exist in debugger context before it is loaded by the
+topology. To run the testbench until breakpoint, select from menu
+Program -> Run... Then mouse copy (text select with left button and
+enter to field with center button) the argument line output from
+previous script run and click "Run".
+
+::
+
+   -d -r 48000 -R 48000 -i audio_in.raw -o audio_out.raw -t ../../build_tools/test/topology/test-playback-ssp5-mclk-0-I2S-eq-iir-s16le-s16le-48k-24576k-codec.tplg -b S16_L
+
+The execution is now stopped to breakpoint. Since the symbols exist
+now the breakpoints can be added to component life cycle after
+new(). Use the lowest window parth with prompt (gdb) for convenience.
+
+.. code-block:: bash
+
+   break eq_iir_cmd
+   break eq_iir_params
+   break eq_iir_prepare
+   break eq_iir_copy
+   break eq_iir_reset
+   break eq_iir_free
+
+After that press "Cont" in the small remote control window next to
+main ddd window. The execution stops to params() function in playback
+start. To view stream parameters mouse left click on params in
+function arguments list and select with right mouse button "Display
+\*params". The same can be done for dev structure. The not shown fiels
+can be expanded and pointers should as pipeline from dev can be looked
+with righ mouse click "Display \*()" from a viewed pointer field. The
+boxes can be moved around with mouse from the borders.
+
+.. figure:: fig_ddd_structs.png
+
+   Viewing data in ddd.
+
+By further pressing "Cont" the code can be run into prepare(). The
+next "Cont" press brings the execution to copy(). A breakpoint can be
+added to known processing function
+
+.. code-block:: bash
+
+   break eq_iir_s32_default
+
+Then in the function after steppping over the read frag operations the
+input and output frames PCM code values for the entire 48 frames to
+process ould be added to debug view with command. Though in this case
+due to x and y updates by get frag operation the base address of x and
+y move.
+
+.. code-block:: bash
+
+   graph display x[0]@96
+   graph display y[0]@96
+
+Or to display the entire sink buffer content to see the circular
+update over two periods of data. The format could be changed to hex if
+desired with right mouse click to data.
+
+.. code-block:: bash
+
+   graph display ((int16_t *)sinkb->stream.addr)[0]@192
+   
+.. note::
+
+   DDD has data plotting capability but at the time of writing this
+   the feature does not seem work. Such can be useful in finding PCM
```

Comment:
the feature -> feature

Reply:
I'd keep "the" here. Though I may be wrong of course.

Reply:
@deb-intel can you fix the grammar as merge. thanks !